### ISSUE TITLE
aria-valuetext not working in IE

### ISSUE THREAD

**Issue by michelhebert** [nvaccess/nvda#6561]
After finding a similar defect at #913, I am opening this defect per @jcsteh 's request. Please note this issue presists in NVDA version 2016.3

Given an HTML5 widget with `role="slider"`, if both `aria-valuenow` and `aria-valuetext` are present, screen readers should announce `aria-valuetext` instead of `aria-valuenow`, as it's the human-readable text alternative of `aria-valuenow`:

http://www.w3.org/TR/wai-aria/states_and_properties#aria-valuetext

NVDA 2016.3 does this currently in Firefox, but ''not'' in IE (tested in IE11). Instead, it reads `aria-valuenow`. IE is apparently exposing `aria-valuetext`, because JAWS 15 and Window-Eyes 8.4 announce it correctly in IE11.

Here are a couple of basic test pages that were created:

http://terrillthompson.com/tests/slider2.html

**Comment by jcsteh** [comment]
From https://github.com/nvaccess/nvda/issues/913#issuecomment-155284762:

> It doesn't expose it via MSAA. It seems it does via UIA, but for various reasons, we can't use that in IE. Therefore, we'll just need to do more DOM parsing (which is almost certainly how other screen readers do it).

What a surprise: IE's a11y implementation is broken in yet another way. :) Anyway... the thing that worries me about this test case is that for some reason, calling node.getAttribute("aria-valuetext") returns null. I don't know why. So, DOM parsing might not be an option.

**Comment by michelhebert** [comment]
This property is used, for example, on a range widget such as a slider or progress bar. 

Instead of the screen-reader just reading the `aria-valuenow` value (which only accepts a decimal), `aria-valuetext` allows web developers to pass a more human readable text alternative.

In my case, I ran into the issue where the slider uses dates.  When it moves, it would spit out "20161108" instead of "November 11, 2016".  With `aria-valuetext`, I can pass human readable text.

Any scenario with sliders/progress bar/spinners would benefit from using `aria-valuetext` so the user knows the unit value as they interact with it, instead of just a decimal, which can be limited in many situations.

examples:
Sliders : https://www.w3.org/TR/wai-aria-practices-1.1/examples/slider/slider-2.html
Spinners (example does not use aria-valuetext, but the point is the usage): http://oaa-accessibility.org/example/33/

**Comment by LaurentGoderre** [comment]
We use Unix timestamps for a date slider and then use the `aria-valuetext` to convert it to a textual representation of the date so this is an issue for us.

**Comment by Adriani90** [comment]
cc: @michaelDCurran

**Comment by Adriani90** [comment]
cc: @seanbudd

**Comment by zstanecic** [comment]
Is it needed to support Ie?

As I know, the Microsoft declared end of support for this piece of soft.

**Comment by Adriani90** [comment]
Support for IE ended completely in 2022 and the browser is automatically removed from machines that still have it after an Edge update in February 2023. Closing this as abandoned since I don't expect any further accessibility updates to IE or to IE mode in Microsoft Edge.
